source/Locales/StoneDuality/ForSpectralLocales.lagda

```diff
@@ -0,0 +1,118 @@
+--------------------------------------------------------------------------------
+title:        Stone duality for spectral locales
+author:       Ayberk Tosun
+date-started: 2024-04-12
+dates-updated: [2024-05-08]
+--------------------------------------------------------------------------------
+
+This module will eventually contain the proof Stone duality for spectral
+locales. It currently contains some preliminary results in this direction.
+
+\begin{code}[hide]
+
+{-# OPTIONS --safe --without-K --lossy-unification --exact-split --double-check --auto-inline #-}
+
+open import MLTT.Spartan hiding (𝟚; ₀; ₁)
+open import UF.FunExt
+open import UF.PropTrunc
+open import UF.Size
+open import UF.Subsingletons
+open import UF.UA-FunExt
+open import UF.Univalence
+
+module Locales.StoneDuality.ForSpectralLocales
+        (ua : Univalence)
+        (pt : propositional-truncations-exist)
+        (sr : Set-Replacement pt)
+       where
+
+private
+ fe : Fun-Ext
+ fe {𝓤} {𝓥} = univalence-gives-funext' 𝓤 𝓥 (ua 𝓤) (ua (𝓤 ⊔ 𝓥))
+
+ pe : Prop-Ext
+ pe {𝓤} = univalence-gives-propext (ua 𝓤)
+
+open import Locales.Compactness pt fe
+open import Locales.ContinuousMap.FrameHomomorphism-Definition pt fe
+open import Locales.ContinuousMap.FrameHomomorphism-Properties pt fe
+open import Locales.ContinuousMap.Homeomorphism-Definition pt fe
+open import Locales.DistributiveLattice.Definition fe pt
+open import Locales.DistributiveLattice.LocaleOfSpectra fe pe pt
+open import Locales.DistributiveLattice.LocaleOfSpectra-Properties fe pe pt
+open import Locales.DistributiveLattice.Resizing ua pt sr
+open import Locales.Frame pt fe
+open import Locales.SIP.FrameSIP
+open import Locales.SmallBasis pt fe sr
+open import Locales.Spectrality.LatticeOfCompactOpens ua pt sr
+open import Locales.Spectrality.SpectralLocale pt fe
+open import Slice.Family
+open import UF.Equiv
+open import UF.Logic
+open import UF.SubtypeClassifier
+
+open AllCombinators pt fe
+open FrameHomomorphismProperties
+open FrameHomomorphisms
+open Locale
+
+\end{code}
+
+We denote by `spec L` the spectrum (the locale defined by the frame of ideals)
+of a distributive lattice `L`.
+
+\begin{code}
+
+open DefnOfFrameOfIdeal
+
+spec : DistributiveLattice 𝓤 → Locale (𝓤 ⁺) 𝓤 𝓤
+spec = locale-of-spectra
```

Comment:
Didn't we discuss that there isn't such a thing as a locale of spectra and agree to change that?

Reply:
Yes, but this is not a change that I made in this PR. This is something already in `master` and I’m merely using it in this PR. I will change it as we discussed, just not in this PR. If I make that change here, the PR will involve many modules that is has nothing to do with. If you’re fine with that, I would be happy to make the change here.